From f62e50b05a1db7a0806dea53af5337fe6e8f79ce Mon Sep 17 00:00:00 2001
From: Roman Stratiienko <r.stratiienko@gmail.com>
Date: Sun, 8 Jan 2023 13:13:54 +0200
Subject: [PATCH] GLODROID: WORKAROUND: Use UTF16BE instead of UCS2BE codepage
 on Android

Android bionic has tiny iconv that contains UTF16, but not UCS2
According to wiki UTF16 is extension of UCS2 and must support UCS2.

Signed-off-by: Roman Stratiienko <r.stratiienko@gmail.com>
---
 src/mm-charsets.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/mm-charsets.c b/src/mm-charsets.c
index 0cbad337..5b2a1e78 100644
--- a/src/mm-charsets.c
+++ b/src/mm-charsets.c
@@ -42,7 +42,12 @@ typedef struct {
 
 static const CharsetSettings charset_settings[] = {
     { MM_MODEM_CHARSET_UTF8,    "UTF-8",   "UTF8",   "UTF-8"     },
+#ifndef __ANDROID__
     { MM_MODEM_CHARSET_UCS2,    "UCS2",    NULL,     "UCS-2BE"   },
+#else
+    /* Android bionic has tiny iconv that contains 'utf16be'. Use it instead. */
+    { MM_MODEM_CHARSET_UCS2,    "UCS2",    NULL,     "UTF-16BE"   },
+#endif
     { MM_MODEM_CHARSET_IRA,     "IRA",     "ASCII",  "ASCII"     },
     { MM_MODEM_CHARSET_GSM,     "GSM",     NULL,     NULL        },
     { MM_MODEM_CHARSET_8859_1,  "8859-1",  NULL,     "ISO8859-1" },
-- 
2.37.2

